### PR TITLE
feat(client): add novu user agent

### PR DESCRIPTION
**Pull request by BiswaViraj**
### What changed? Why was the change needed?

- adds custom `novu-user-agent` header to monitor packages using `@novu/client`

### Screenshots

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Special notes for your reviewer

</details>

### CODE REVIEW

File: packages/client/src/http-client/http-client.ts [hunk 1]
```diff
@@ -11,11 +15,13 @@ export class HttpClient {
   constructor({
     apiVersion = DEFAULT_API_VERSION,
     backendUrl = DEFAULT_BACKEND_URL,
+    novuUserAgent = NOVU_USER_AGENT,
   }: ApiOptions = {}) {
     this.apiVersion = apiVersion;
     this.backendUrl = `${backendUrl}/${this.apiVersion}`;
     this.headers = {
       'Content-Type': 'application/json',
+      'X-Novu-User-Agent:': novuUserAgent,
```
Comment (SokratisVidros) [inline packages/client/src/http-client/http-client.ts]:
```suggestion
      'User-Agent:': novuUserAgent,
```

File: packages/client/src/index.ts [hunk 2]
```diff
@@ -53,6 +53,7 @@ export type PreferenceSettingsType = {
 export type ApiOptions = {
   apiVersion?: string;
   backendUrl?: string;
+  novuUserAgent?: string;
```
Comment (SokratisVidros) [inline packages/client/src/index.ts]:
```suggestion
  userAgent?: string;
```

**Review by SokratisVidros — CHANGES_REQUESTED** [review]

**Review by SokratisVidros — APPROVED** [review]

File: packages/client/src/index.ts (hunk 2 above)
Comment (scopsy) [reply packages/client/src/index.ts]: **question:** a small question, shouldn't we keep it consistent here with camelCasing? for public facing api's? 

I believe it should match the `constructor` signature above?  
```
 constructor({
    apiVersion = DEFAULT_API_VERSION,
    backendUrl = DEFAULT_BACKEND_URL,
    novuUserAgent = NOVU_USER_AGENT,
  }: ApiOptions = {}) {
```

Comment (SokratisVidros) [reply packages/client/src/index.ts]: Of course. I blame Grammarly for this 😆

**Review by SokratisVidros — APPROVED** [review]

**Pull request merged**